src/vscode-dts/vscode.proposed.lmTools.d.ts

```diff
@@ -91,7 +91,32 @@ declare module 'vscode' {
 		 * Invoke a tool with the given parameters.
 		 * TODO@API Could request a set of contentTypes to be returned so they don't all need to be computed?
 		 */
-		export function invokeTool(id: string, parameters: Object, token: CancellationToken): Thenable<LanguageModelToolResult>;
+		export function invokeTool(id: string, options: LanguageModelToolInvokationOptions, token: CancellationToken): Thenable<LanguageModelToolResult>;
+	}
+
+	export interface LanguageModelToolInvokationOptions {
+		/**
+		 * Parameters with which to invoke the tool.
+		 */
+		parameters: Object;
+
+		/**
+		 * Options to hint at how many tokens the tool should return in its response.
+		 */
+		tokenOptions?: {
```

Comment:
Should this be optional when calling `lm.invokeTool` but somehow be set when receiving it in `LanguageModelTool#invoke`? And/or when calling `lm.invokeTool` you might wanna pass a `LanguageModelChat`-instance so that we do the plumbing of the token count function?

Reply:
This also makes a case to have two distinct types for `lm.invokeTool` and `LanguageModelTool#invoke`

Reply:
I don't think we can always depend on a consumer knowing ahead of time what tokens to use / how to count the tokens. At the moment the LM tools API can technically be called by any extension without having to necessarily use the vscode chat API or be backed by a LM at all (e.g. no need to have a LanguageModelChat). I think that lack of coupling may be useful long term.